scripts/library.sh

```diff
@@ -332,7 +332,15 @@ function start_latest_knative_serving() {
   header "Starting Knative Serving"
   subheader "Installing Knative Serving"
   echo "Installing Serving from ${KNATIVE_SERVING_RELEASE}"
-  kubectl apply -f ${KNATIVE_SERVING_RELEASE} || return 1
+  # Some CRDs defined in serving YAML are also referenced by other components in serving. As it takes
+  # time for CRDs to become effective, there is a race condition between when the CRDs are effective
+  # and when the resources that references those CRDs are created.
+  # The current workaround is to re-apply serving.yaml if it fails. Remove the retry logic after the
+  # race condition is fixed. (https://github.com/knative/serving/issues/4176)
+  if [[ `kubectl apply -f ${KNATIVE_SERVING_RELEASE}` ]]; then
+    sleep 60
```

Comment:
I would also add a

```
echo "Install failed, waiting 60s and then retrying..."
```

here so it's clear in the logs what's going on.

Reply:
Thanks. Missed the log statement after moving the retry out of the function.